client/layout/masterbar-item-notifications.jsx

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDom from 'react-dom';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import MasterbarItem from './masterbar-item';
+import Notifications from 'notifications';
+import store from 'store';
+
+export default React.createClass( {
+	displayName: 'MasterbarItemNotifications',
+
+	propTypes: {
+		user: React.PropTypes.object,
+		isActive: React.PropTypes.bool,
+		className: React.PropTypes.string,
+		onClick: React.PropTypes.func,
+	},
+
+	getInitialState() {
+		let newNote = false;
+		let user;
+
+		if ( this.props.user ) {
```

Comment:
I would probably add `.isRequired` on the end of the `propType` and remove the check, just because it gives someone studying this code one thing less to think about. But it would be fine as it is.